Packages/Netherlands3D/WebServiceReader/Runtime/Scripts/CreateWMSLayer.cs

```diff
@@ -50,135 +44,77 @@ public class WMSLOD
             new WMSLOD() { textureSize = 2048, maximumDistance = 1000 }
         };
 
-        [SerializeField]
-        private int tileSize = 1500;
+        [SerializeField] private int tileSize = 1500;
         [SerializeField] private bool compressLoadedTextures = true;
-        private bool OnlyOnTerrain_Memory = false;
         private bool DisplayState = true;
 
         void Start()
         {
-            tileHandler = FindObjectOfType(typeof(TileSystem.TileHandler)) as TileSystem.TileHandler;
-            if (OnWMSUrlDefined_String)
-            {
-                OnWMSUrlDefined_String.AddListenerStarted(CreateWMS_Layer);
-            }
-            if (ShowWMSOnBuildings)
-            {
-                ShowWMSOnBuildings.AddListenerStarted(showWMSOnBuildings);
-            }
-            if (ShowWMSOnTerrain)
-            {
-                ShowWMSOnTerrain.AddListenerStarted(showWMSOnTerrain);
-            }
-            if (ShowWMSOnBuildingsAndTerrain)
-            {
-                ShowWMSOnBuildingsAndTerrain.AddListenerStarted(showWMSOnBuildingsAndTerrain);
-            }
-            if (UnloadWMSService)
-            {
-                UnloadWMSService.AddListenerStarted(UnloadLayer);
-            }
-            if (AleenOpMaaiveld_Bool)
-            {
-                AleenOpMaaiveld_Bool.AddListenerStarted(ShowOnlyOnTerrain);
-            }
-            if (ShowLayer_Bool)
-            {
-                ShowLayer_Bool.AddListenerStarted(ShowLayer);
-            }
+            tileHandler = FindObjectOfType(typeof(TileHandler)) as TileHandler;
+            if (!tileHandler)
```

Comment:
Goede toevoeging, ik vraag mij wel af waarom het een impliciete dependency is en niet een Serialized Field?

Reply:
Yo Mike. Goede vraag. Voornamelijk om te onmoedigen zo'n referentie in de scene op te slaan, en niet op te leggen dat de tilehandler op het zelfde object als dit script moet zitten, maar bedenk me wel dat er theoretisch meerdere TileHandlers zouden kunnen bestaan in een project; dus zal t veld serializen, en alleen opzoeken wanneer hij null is bij Start